react/features/stream-effects/noise-suppression/NoiseSuppressionEffect.ts

```diff
@@ -0,0 +1,83 @@
+// @ts-ignore
+import { getBaseUrl } from '../../base/util';
+
+/**
+ * Class Implementing the effect interface expected by a JitsiLocalTrack.
+ * Effect applies rnnoise denoising on a audio JitsiLocalTrack.
+ */
+export class NoiseSuppressionEffect {
+
+    /**
+     * Web audio context.
+     */
+    private _audioContext: AudioContext;
+
+    /**
+     * Source that will be attached to the track affected by the effect.
+     */
+    private _audioSource: MediaStreamAudioSourceNode;
+
+    /**
+     * Destination that will contain denoised audio from the audio worklet.
+     */
+    private _audioDestination: MediaStreamAudioDestinationNode;
+
+    /**
+     * `AudioWorkletProcessor` associated node.
+     */
+    private _noiseSuppressorNode: AudioWorkletNode;
+
+    /**
+     * Effect interface called by source JitsiLocalTrack.
+     * Applies effect that uses a {@code NoiseSuppressor} service initialized with {@code RnnoiseProcessor}
+     * for denoising.
+     *
+     * @param {MediaStream} audioStream - Audio stream which will be mixed with _mixAudio.
+     * @returns {MediaStream} - MediaStream containing both audio tracks mixed together.
+     */
+    startEffect(audioStream: MediaStream) : MediaStream{
+        this._audioContext = new AudioContext();
+
+        this._audioSource = this._audioContext.createMediaStreamSource(audioStream);
+        this._audioDestination = this._audioContext.createMediaStreamDestination();
+
+        const baseUrl = `${getBaseUrl()}libs/`;
+        const workletUrl = `${baseUrl}noise-suppressor-worklet.min.js`;
+
+        // Connect the audio processing graph MediaStream -> AudioWorkletNode -> MediaStreamAudioDestinationNode
+        this._audioContext.audioWorklet.addModule(workletUrl).then(() => {
```

Comment:
What if this fails? I feel like we need a `catch` here, and potentially some logging / handling of the problem.

Reply:
startEffect is called by the JitsiLocalTrack and any potential exceptions should be bubbled up and handled [here ](https://github.com/jitsi/lib-jitsi-meet/blob/634885b9fb513103a690b43396d8369544b50cf3/modules/RTC/JitsiLocalTrack.js#L881,) or at least that's my understanding of it.
Idk if there's anything actionable at this point in case of an exception, so I just let the JitsiLocalTrack handle the logging.

Reply:
But the failure may happen later, since it's a promise chain.

Reply:
I see, will add a catch with a log there.